packages/babel-plugin/src/utils/css-builders.tsx

```diff
@@ -400,6 +400,58 @@ const extractLogicalExpression = (node: t.ArrowFunctionExpression, meta: Metadat
   return { css: mergeSubsequentUnconditionalCssItems(css), variables };
 };
 
+/**
+ * Manipulates the AST to ensure that CSS variables are generated correctly for negative values
+ *
+ * consider we have the following:
+ *
+ * const gridSize = 8;
+ *
+ * const LayoutRight = styled.aside`
```

Comment:
Does `margin-left: -${gridSize}px;` work?

Reply:
good catch

Reply:
added logic to handle this